core/dmux.go

```
@@ -311,7 +378,172 @@ func simpleSetup(size, qsize int, sink Sink) ([]chan interface{}, *sync.WaitGrou
+						ConsumerGroupName: sideline.ConsumerGroupName,
+						ClusterName:       sideline.ClusterName,
+						Message:           val.GetRawMsg().Value,
+						Version:           channelObject.Version,
+						ConnectionType:    sideline.ConnectionType,
+						SidelineMeta:      sidelineMetaByteArray,
+					}
+
+					sidelineByteArray, err := json.Marshal(kafkaSidelineMessage)
+					if err != nil {
+						log.Printf("error in serde of kafkaSidelineMessage")
+						continue
+					}
+					sidelineMessageResponse := sidelineImpl.(sideline_models.CheckMessageSideline).SidelineMessage(sidelineByteArray)
+					if !sidelineMessageResponse.Success {
+						var check sideline_models.CheckMessageSidelineResponse
+						log.Printf(sidelineMessageResponse.ErrorMessage)
+						if sidelineMessageResponse.ConcurrentModificationError {
+							checkSidelineMessage := sideline_models.SidelineMessage{
+								GroupId:           string(val.GetRawMsg().Key),
+								Partition:         val.GetRawMsg().Partition,
+								EntityId:          string(val.GetRawMsg().Key) + sideline.ConsumerGroupName + sideline.ClusterName,
+								Offset:            val.GetRawMsg().Offset,
+								ConsumerGroupName: sideline.ConsumerGroupName,
+								ClusterName:       sideline.ClusterName,
+								Message:           val.GetRawMsg().Value,
+								ConnectionType:    sideline.ConnectionType,
+							}
+							checkSidelineMessageBytes, checkSidelineMessageErr := json.Marshal(checkSidelineMessage)
+							if checkSidelineMessageErr != nil {
+								log.Printf("error in serde of checkSidelineMessage")
+								errors.New("error in serde of checkSidelineMessage")
+								continue
+							}
+							checkBytes, checkErr := sidelineImpl.(sideline_models.CheckMessageSideline).
+								CheckMessageSideline(checkSidelineMessageBytes)
+							err := json.Unmarshal(checkBytes, &check)
+							if err != nil {
+								errors.New("error in serde of CheckMessageSidelineResponse")
```

This error is not being assigned. Also its usually not right to swallow the original err. Returning the actual error would be helpful in debugging/understanding the error.

---

will change this